src/app/shared/filter-by-property/filter-by-property.pipe.ts

```diff
@@ -0,0 +1,28 @@
+import {Pipe, PipeTransform} from '@angular/core';
+
+@Pipe({
+    name: 'filterByProperty',
+})
+export class FilterByPropertyPipe implements PipeTransform {
+    transform<T, P extends keyof T>(
+        items: T[] | undefined | null,
+        propertyName: P,
+        searchPropertyValue: T[P],
+    ): T[] | undefined | null {
+        if (!items || items.length === 0) {
```

Comment:
Я бы предложил сократить проверку, в читаемости на мой взгляд не теряет:
```suggestion
        if (!items?.length) {
```